quickstart/aspnet/ContosoTeamStats/Controllers/HomeController.cs

```diff
@@ -83,5 +97,138 @@ public static ConnectionMultiplexer Connection
+                    previousErrorTime = utcNow;
+                    return;
+                }
+
+                if (elapsedSinceLastReconnect < ReconnectMinFrequency)
+                    return; // Some other thread made it through the check and the lock, so nothing to do.
+
+                TimeSpan elapsedSinceFirstError = utcNow - firstErrorTime;
+                TimeSpan elapsedSinceMostRecentError = utcNow - previousErrorTime;
+
+                bool shouldReconnect =
+                    elapsedSinceFirstError >= ReconnectErrorThreshold // Make sure we gave the multiplexer enough time to reconnect on its own if it could.
+                    && elapsedSinceMostRecentError <= ReconnectErrorThreshold; // Make sure we aren't working on stale data (e.g. if there was a gap in errors, don't reconnect yet).
+
+                if (!shouldReconnect)
+                    return;
+
+                // Update the previousErrorTime timestamp to be now (e.g. this reconnect request).
+                previousErrorTime = utcNow;
+
+                firstErrorTime = DateTimeOffset.MinValue;
+                previousErrorTime = DateTimeOffset.MinValue;
+
+                Lazy<ConnectionMultiplexer> oldConnection = lazyConnection;
+                CloseConnection(oldConnection);
+                lazyConnection = CreateConnection();
+                Interlocked.Exchange(ref lastReconnectTicks, utcNow.UtcTicks);
+            }
+        }
+
+        public static IDatabase GetDatabase()
+        {
+            try
+            {
+                return Connection.GetDatabase();
+            }
+            catch (Exception ex) when (ex is RedisConnectionException || ex is SocketException)
+            {
+                ForceReconnect();
+                return GetDatabase();
+            }
+            catch (ObjectDisposedException)
+            {
+                return GetDatabase();
+            }
+        }
+
+        public static System.Net.EndPoint[] GetEndPoints()
+        {
+            try
+            {
+                return Connection.GetEndPoints();
+            }
+            catch (Exception ex) when (ex is RedisConnectionException || ex is SocketException)
+            {
+                ForceReconnect();
+                return GetEndPoints();
+            }
+            catch (ObjectDisposedException)
+            {
+                return GetEndPoints();
+            }
+        }
+
+        public static IServer GetServer(string host, int port)
+        {
+            try
+            {
+                return Connection.GetServer(host, port);
+            }
+            catch (Exception ex) when (ex is RedisConnectionException || ex is SocketException)
+            {
+                ForceReconnect();
+                return GetServer(host, port);
+            }
+            catch (ObjectDisposedException)
+            {
+                return GetServer(host, port);
+            }
+        }
```

Comment:
@philon-msft, thanks a bunch for the speedy review! One thing I should probably run by you for a double-check is if these retry methods look OK, at least as starting point for now? The original gist and the earlier dotnet/ClientSamples/StackExchange.Redis sample didn't yet include an example of what to do after calling `ForceReconnect()`, but I think these methods are roughly correct? The approach is inspired by [a comment on the gist](https://gist.github.com/JonCole/925630df72be1351b21440625ff2671f#gistcomment-3071233):

_**JonCole** commented on Oct 31, 2019_

> > This guide is really amazing, I am just wondering what the best way of handling the ObjectDisposedException is?
> 
> If I were to build this, I would create a wrapper around the IConnectionMultiplexer/IDatabase interfaces that handles the ObjectDisposedExceptions that are thrown, such that as few callers have to be aware of this as possible. I expect that when you see these errors, retrying immediately on the new connection should be fine.

Reply:
Those look like a good starting point to me. You may want to guard against a stack overflow in the case of prolonged outage, by only catching and retrying each method 5ish times, and just give up and rethrow after that.

Reply:
Great idea. I adjusted the quickstarts to take that approach with a default of 5 retries.